bokeh/util/serialization.py

```diff
@@ -55,7 +55,7 @@ def transform_series(obj):
     return transform_array(vals)
 
 def transform_array_to_list(array):
-    if np.isnan(array).any() or np.isinf(array).any():
+    if (array.dtype.kind in ('u', 'i', 'f') and (~np.isfinite(array)).any()):
```

Comment:
I've removed the other condition so this seems right to me, if it's numeric and any value is non-finite replace the non-finite values.